Sources/App/Utilities/Environment+dotenv.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copy of extension from vapor-community/vapor-ext
+// https://github.com/vapor-community/vapor-ext/blob/master/Sources/ServiceExt/Environment%2BDotEnv.swift
+//
+import Foundation
+import Service
+
+#if os(Linux)
+    import Glibc
+#else
+    import Darwin
+#endif
+
+// MARK: - Methods
+
+public extension Environment {
+    /// Loads environment variables from .env files.
+    ///
+    /// - Parameter filename: name of your env file.
+    static func dotenv(filename: String = ".env") {
+        guard let path = getAbsolutePath(for: filename),
+            let contents = try? String(contentsOfFile: path, encoding: .utf8) else {
+            return
+        }
+
+        let lines = contents.split(whereSeparator: { $0 == "\n" || $0 == "\r\n" })
+
+        for line in lines {
+            // ignore comments
+            if line.starts(with: "#") {
+                continue
+            }
+
+            // ignore lines that appear empty
+            if line.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+                continue
+            }
+
+            // extract key and value which are separated by an equals sign
+            let parts = line.components(separatedBy: "=")
```

Comment:
As you said, we know that this causes problems with values containing `=` signs. Here's how we could fix that:

```swift
guard
    let indexOfKeyValueSeparator = line.firstIndex(of: "="), // has to contain at least one "="
    indexOfKeyValueSeparator != line.startIndex // "=" must not be first character
else {
    continue
}

let key = line.prefix(upTo: indexOfKeyValueSeparator)
let value = line.suffix(from: line.index(after: indexOfKeyValueSeparator))
```

Reply:
We must return in a guard's else block, right?

Reply:
You must exit the current scope. That can be `return` (in a function), `continue` (in a loop), `throw ...`, or `fatalError()` and maybe others I can't think of right now

Reply:
Alright updated this